File: doc-examples/nitrite-example-java/src/main/java/example/Customer.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

// tag::customer[]
@MappedEntity
public class Customer {
    @Id
    @GeneratedValue
    private String id;

    private String name;
    private Address address;

    public Customer(String name, Address address) {
        this.name = name;
        this.address = address;
    }
    // end::customer[]

    public Customer() {
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }
// tag::customer[]
}
// end::customer[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/Customer.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity

// tag::customer[]
@MappedEntity
class Customer {
    @Id
    @GeneratedValue
    var id: String? = null

    var name: String? = null
    var address: Address? = null

    constructor(name: String, address: Address) {
        this.name = name
        this.address = address
    }
    // end::customer[]

    constructor()
// tag::customer[]
}
// end::customer[]
